### PR TITLE
Fix memory usage in catalog operator

### PR DESCRIPTION
**Pull request by ecordell**
Also scopes the catalog operator to only watch the "catalogNamespace" for catalog sources (it was watching all before).

Mem usage before:
screen shot 2018-05-31 at 9 26 34 am

Mem usage after:
screen shot 2018-05-31 at 6 12 54 pm

### CODE REVIEW

**Comment by ecordell** [comment]
(Pipeline is passing now, status is just slow to update)

File: pkg/controller/operators/catalog/operator.go [hunk 1]
```diff
@@ -69,6 +67,13 @@ func NewOperator(kubeconfigPath string, wakeupInterval time.Duration, operatorNa
 		subSharedIndexInformers = append(subSharedIndexInformers, nsInformerFactory.Subscription().V1alpha1().Subscriptions().Informer())
 	}
 
+	// Create an informer for each catalog namespace
+	catsrcSharedIndexInformers := []cache.SharedIndexInformer{}
+	for _, namespace := range []string{operatorNamespace} {
```
Comment (alecmerdler) [inline pkg/controller/operators/catalog/operator.go]: Why `range` over a slice with a single item?

File: pkg/controller/operators/catalog/operator.go [hunk 2]
```diff
@@ -139,7 +142,7 @@ func (o *Operator) syncCatalogSources(obj interface{}) (syncError error) {
 	defer o.sourcesLock.Unlock()
 	o.sources[catsrc.GetName()] = src
 	o.sourcesLastUpdate = timeNow()
-	return err
+	return nil
```
Comment (alecmerdler) [inline pkg/controller/operators/catalog/operator.go]: We mix this convention in the codebase (we know `err == nil`, but return `err` anyway). Which is preferred?

File: pkg/controller/registry/configmap_loader.go [hunk 3]
```diff
@@ -21,23 +21,22 @@ const (
 
 // ConfigMapCatalogResourceLoader loads a ConfigMap of resources into the in-memory catalog
 type ConfigMapCatalogResourceLoader struct {
-	Catalog   *InMem
```
Comment (alecmerdler) [inline pkg/controller/registry/configmap_loader.go]: Why the change in this interface?

File: pkg/controller/operators/catalog/operator.go (hunk 1 above)
Comment (ecordell) [reply pkg/controller/operators/catalog/operator.go]: Just because I know we'll implement "local" catalogs in the near future, where we'll want to iterate over `operatorNamespace ∪ watchedNamespaces` instead of just `operatorNamespace`

File: pkg/controller/operators/catalog/operator.go (hunk 2 above)
Comment (ecordell) [reply pkg/controller/operators/catalog/operator.go]: I feel this is clearer, but I don't know if there's a golang convention for this. Which do you think is more readable?

File: pkg/controller/registry/configmap_loader.go (hunk 3 above)
Comment (ecordell) [reply pkg/controller/registry/configmap_loader.go]: This is fixing the memory leak, since the catalog returned was a field of the loader, and the loader holds a reference to part of the operator (operator client), every time we sync the catalog we were loading into memory and retaining it indefinitely. 

We might still be leaking the loader itself now that you mention it...but at least that's a tiny amount instead of an entire in memory catalog
